src/main/java/uk/gov/cshr/controller/reset/ResetController.java

```diff
@@ -0,0 +1,131 @@
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private ResetFormValidator resetFormValidator;
+
+    @GetMapping
+    public String reset() {
+        return "user-reset";
+    }
+
+    @PostMapping
+    public String requestReset(@RequestParam(value = "email") String email) throws Exception {
+        LOGGER.info("Requesting reset for {} ", email);
+
+        if (identityRepository.existsByEmail(email)) {
+            resetService.notifyForResetRequest(email);
+        }
+
+        return "user-checkEmail";
+    }
+
+    @GetMapping("/{code}")
+    public String loadResetForm(@PathVariable(value = "code") String code, RedirectAttributes redirectAttributes, Model model) {
+        LOGGER.info("User on reset screen with code {}", code);
+
+        Reset reset = resetRepository.findByCode(code);
+
+        if (!resetRepository.existsByCode(code)) {
+            LOGGER.info("{} reset code does not exist", code);
+            redirectAttributes.addFlashAttribute("status", "There was an error with your reset, please try again.");
+            return "redirect:/login";
+        }
+
+        if (resetService.isResetExpired(reset) || !resetService.isResetValid(reset)) {
+            LOGGER.info("Reset is not valid for code {}", code);
+            return "redirect:/reset";
+        }
+
+        ResetForm resetForm = new ResetForm();
+        resetForm.setCode(code);
+
+        model.addAttribute("resetForm", resetForm);
+
+        return "user-passwordForm";
+    }
+
+    @PostMapping("/{code}")
+    public String resetPassword(@PathVariable(value = "code") String code, @ModelAttribute @Valid ResetForm resetForm, BindingResult bindingResult, Model model) throws NotificationClientException {
+
+        if (bindingResult.hasErrors()) {
+
+            model.addAttribute("resetForm", resetForm);
+            return "user-passwordForm";
+        }
+        Reset reset = resetRepository.findByCode(code);
+
+        if (resetService.isResetExpired(reset) || !resetService.isResetValid(reset)) {
+            LOGGER.info("Reset is not valid for code {}", code);
+            return "redirect:/reset";
+        }
+
+        if (reset == null || reset.getEmail() == null) {
+            LOGGER.info("Reset does not exist with code {}", code);
+            return "redirect:/reset";
+        }
+
+        Identity identity = identityRepository.findFirstByActiveTrueAndEmailEquals(reset.getEmail());
+
+        if (!reset.getEmail().equals(identity.getEmail())) {
+            LOGGER.info("Reset email and Identity email do not match for code {}, and email", code, identity.getEmail());
+            return "redirect:/reset";
+        }
+
+        if (identity == null || identity.getEmail() == null) {
+            LOGGER.info("Identity does not exist with reset code {}", code);
+            return "redirect:/reset";
+        }
+
+        identity.setPassword(passwordEncoder.encode(resetForm.getPassword()));
```

Comment:
MAINTENANCE - please refactor to `identityService.updatePassword(identity, password)`